### DISCUSSION TITLE
The 'tab' is not convenient

### DISCUSSION THREAD

**Discussion by x2c3z4** [adoyle-h/one.nvim#18] [Q&A] [answered]
How could I get this tab style, like vscode?
![image](https://github.com/adoyle-h/one.nvim/assets/935511/d6a010be-f5a2-4b1c-8814-6a313dc55ef2)

Currently the 'tab' navigation is on the right cornor, and no file name.

**Comment by adoyle-h** [comment, marked as answer]
Current one.nvim use [tabby.nvim](https://github.com/nanozuki/tabby.nvim) as tabline. [Current config](https://github.com/adoyle-h/one.nvim/blob/master/lua/one/plugins/tabby.lua) is my favorite.

The tabby.nvim is highly configurable. You can override the config as you like. For example, below config comes from [tabby.nvim document](https://github.com/nanozuki/tabby.nvim#setup).

```lua
require('one').setup {
  plugins = {

		{
			'nanozuki/tabby.nvim',

			config = function()
				vim.o.showtabline = 2

				-- see https://github.com/nanozuki/tabby.nvim#setup
				local theme = {
					fill = 'TabLineFill',
					-- Also you can do this: fill = { fg='#f2e9de', bg='#907aa9', style='italic' }
					head = 'TabLine',
					current_tab = 'TabLineSel',
					tab = 'TabLine',
					win = 'TabLine',
					tail = 'TabLine',
				}

				require('tabby.tabline').set(function(line)
					return {
						{ { '  ', hl = theme.head }, line.sep('', theme.head, theme.fill) },
						line.tabs().foreach(function(tab)
							local hl = tab.is_current() and theme.current_tab or theme.tab
							return {
								line.sep('', hl, theme.fill),
								tab.is_current() and '' or '󰆣',
								tab.number(),
								tab.name(),
								tab.close_btn(''),
								line.sep('', hl, theme.fill),
								hl = hl,
								margin = ' ',
							}
						end),
						line.spacer(),
						line.wins_in_tab(line.api.get_current_tab()).foreach(function(win)
							return {
								line.sep('', theme.win, theme.fill),
								win.is_current() and '' or '',
								win.buf_name(),
								line.sep('', theme.win, theme.fill),
								hl = theme.win,
								margin = ' ',
							}
						end),
						{ line.sep('', theme.tail, theme.fill), { '  ', hl = theme.tail } },
						hl = theme.fill,
					}
				end)
			end,
		},

  }
}
```

It will look like that. The tabs on left, and wins_in_tab on right.

![CleanShot 2023-07-01 at 18 44 52@2x](https://github.com/adoyle-h/one.nvim/assets/1998490/8e73ac01-4fd0-48f9-9ba8-0aee0cef0278)

**Reply by x2c3z4** [reply]
Thanks, it works.
Going further, how to make goto-preview jump to a new tab instead of reusing the current tab.
Currently I have to press 'gd' to open the float window, then press 'c-w t'.

**Reply by adoyle-h** [reply]
Using the [`post_open_hook`](https://github.com/rmagatti/goto-preview#%EF%B8%8F-configuration) function.

> The post_open_hook function gets called right before setting the cursor position in the new floating window. 



Try this config.

```
require('one').setup {
	config = {
		lsp = {
			preview = {
				post_open_hook = function()
					vim.api.nvim_feedkeys(vim.api.nvim_replace_termcodes('<C-w>T', true, false, true), 'n', false)
				end,
			},
		},
	},
}
```

**Reply by x2c3z4** [reply]
Looks good, Thanks.
